pyro/contrib/gp/models/gpr.py

```diff
@@ -44,7 +44,7 @@ def model(self):
         noise = self.get_param("noise")
 
         K = kernel(self.X) + noise.expand(self.num_data).diag()
-        zero_loc = torch.tensor(K.data.new([0])).expand(self.num_data)
+        zero_loc = torch.zeros_like(self.y)
```

Comment:
You might change `y` shape like this: `y = self.y if self.y.dim() == 1 else self.y.t()`